src/components/PurchaseOrder/POForm.js

```diff
@@ -173,6 +173,15 @@ class POForm extends Component {
       this.getLastMenu('clickable-create-new-purchase-order', 'ui-orders.paneMenu.saveOrder');
     const showDeleteButton = initialValues.id || false;
     const orderNumberSetting = getOrderNumberSetting(get(parentResources, 'orderNumberSetting.records', []));
+    const addresses = get(parentResources, 'addresses.records', []).map(address => {
+      const value = JSON.parse(address.value);
```

Comment:
`JSON.parse` might raise an exception. Maybe it's better to leverage this logic in an util function with handling exception

Reply:
Good point. Done